bqn-comint.el

```diff
@@ -101,28 +79,42 @@ If it doesn't exist, create and return it; else, return the existing one."
       (switch-to-buffer-other-window *bqn-process-buffer-name*)
       (bqn-inferior-mode)
       (set-input-method "BQN-Z"))))
-
-(defun bqn-process-execute-region (start end &optional dont-follow)
-  "Send the current region to the bqn-process-session.
-
-When DONT-FOLLOW is non-nil, maintain focus on the buffer where the function was called from."
+(defun bqn-process-execute-region (start end &optional dont-follow return-output)
   (interactive "r")
-  (when (= start end)
-    (error
-     (concat "Attempt to send empty region to "
-             *bqn-process-buffer-name*)))
-  (when bqn-flash-on-send
-    (bqn--flash-region start end))
   (let ((region (buffer-substring-no-properties start end))
         (session (bqn-process-ensure-session))
         (buffer (current-buffer)))
-    (pop-to-buffer (process-buffer session))
-    (goto-char (point-max))
-    (insert (format "\n%s\n" region))
+    (bqn-process-execute-region-setup start end session region)
+    (bqn-process-execute-region-execute buffer)))
+(defun bqn-process-execute-region-setup (start end session region)
```

Comment:
If the function should be hidden, not meant to be consumed by the end user, prefix it with double-hyphen, `bqn--`

https://github.com/bbatsov/emacs-lisp-style-guide#naming